src/browser/components/TabBar.jsx

```diff
@@ -2,12 +2,17 @@
 // Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 import React from 'react';
+import {remote} from 'electron';
 import PropTypes from 'prop-types';
-import {Glyphicon, Nav, NavItem} from 'react-bootstrap';
+import {Nav, NavItem} from 'react-bootstrap';
+import {Container, Draggable} from 'react-smooth-dnd';
+import PlusIcon from 'mdi-react/PlusIcon';
 
 export default class TabBar extends React.Component { // need "this"
   render() {
-    const tabs = this.props.teams.map((team, index) => {
+    const orderedTabs = this.props.teams.concat().sort((a, b) => a.order - b.order);
+    const tabs = orderedTabs.map((team) => {
+      const index = this.props.teams.indexOf(team);
```

Comment:
you can get the index out of the map function as a parameter: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map